lib/account/widgets/avatar.dart

```diff
@@ -0,0 +1,125 @@
+import 'package:flutter/material.dart';
+import 'package:motis_mitfahr_app/account/pages/avatar_picture_page.dart';
+import 'package:image_picker/image_picker.dart';
+import 'package:supabase_flutter/supabase_flutter.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+import '../../util/supabase.dart';
+import '../models/profile.dart';
+
+class Avatar extends StatefulWidget {
+  const Avatar(
+    this.profile, {
+    super.key,
+    this.isTappable = false,
+    this.onUpload,
+    this.size,
+  });
+
+  final bool isTappable;
+  final Profile profile;
+  final Function? onUpload;
+  final double? size;
+
+  @override
+  AvatarState createState() => AvatarState();
+}
+
+class AvatarState extends State<Avatar> {
+  bool _isLoading = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return Stack(
+      children: [
+        Hero(
+          tag: "Avatar-${widget.profile.id}",
+          child: CircleAvatar(
+            radius: widget.size,
+            backgroundImage: widget.profile.avatarUrl?.isEmpty ?? true ? null : NetworkImage(widget.profile.avatarUrl!),
+            child: widget.profile.avatarUrl?.isEmpty ?? true
+                ? Text(
+                    widget.profile.username[0].toUpperCase(),
+                    style: TextStyle(fontSize: widget.size),
+                  )
+                : null,
+          ),
+        ),
+        if (widget.isTappable)
+          Positioned.fill(
+            child: Material(
+              color: Colors.transparent,
+              child: Semantics(
+                label: S.of(context).widgetAvatarDetailsLabel,
+                button: true,
+                tooltip: S.of(context).widgetAvatarDetailsTooltip,
+                child: InkWell(
+                  borderRadius: BorderRadius.circular(widget.size ?? 20),
+                  onTap: () => Navigator.of(context).push(
+                    MaterialPageRoute(
+                      builder: (context) => AvatarPicturePage(widget.profile),
+                    ),
+                  ),
+                ),
+              ),
+            ),
+          ),
+        if (widget.onUpload != null && widget.profile.isCurrentUser)
+          Positioned(
+            right: 0,
+            bottom: 0,
+            child: CircleAvatar(
+              radius: 20,
+              backgroundColor: Theme.of(context).colorScheme.primary,
+              child: IconButton(
+                tooltip: S.of(context).widgetAvatarUploadTooltip,
+                iconSize: 20,
+                onPressed: _isLoading ? null : _upload,
+                icon: const Icon(Icons.camera_alt),
```

Comment:
Icons.camera_alt as an icon gave me the intuition that it is for taking a picture with the kamera and then to put it as a Avatar.
Icons.photo represents the funktion of picking a picture out of photo better.

Reply:
I agree. There is actually also Icons.photo_library, which is even more appropriate